specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,1026 @@
+
+        // Removes a toast having a specific tag
+        Windows.Foundation.IAsyncAction RemoveWithTagAsync(String tag);
+
+        // Removes a toast having a specific tag and group
+        Windows.Foundation.IAsyncAction RemoveWithTagGroupAsync(String tag, String group);
+
+        // Remove all toasts for a specific group
+        Windows.Foundation.IAsyncAction RemoveGroupAsync(String group);
+
+        // Removes all the toasts for the App from Action Centre
+        Windows.Foundation.IAsyncAction RemoveAllAsync();
+
+        // Gets all the toasts for the App from Action Centre
+        Windows.Foundation.IAsyncOperation<Windows.Foundation.Collections.IVector<ToastNotification> > GetAllAsync();
+    };
+
+    [experimental]
+    // The manager class which encompasses all Toast API Functionality
+    runtimeclass ToastNotificationManager
+    {
+        // Gets a Default instance of a ToastNotificationManager
+        static ToastNotificationManager Default{ get; };
+
+        // Register an activator using an ActivationInfo context and caches the token for unregister
+        void RegisterActivator(ToastActivationInfo details);
+
+        // Unregisters the activator and removes the cached Registration token.
+        void UnregisterActivator();
+
+        // Event handler for Toast Activations
+        event Windows.Foundation.EventHandler<ToastActivatedEventArgs> ToastActivated;
+
+        // Displays the Toast in Action Centre
+        void ShowToast(ToastNotification toast);
+
+        // Updates the Toast for a Progress related operation using Tag and Group
+        Windows.Foundation.IAsyncOperation<ToastProgressResult> UpdateToastProgressDataAsync(ToastProgressData data, String tag, String group);
+
+        // Updates the Toast for a Progress related operation using Tag
+        Windows.Foundation.IAsyncOperation<ToastProgressResult> UpdateToastProgressDataAsync(ToastProgressData data, String tag);
+
+        // Get the Toast Setting status for the app
+        ToastNotificationSetting Setting{ get; };
+
+        // Gets an instance of ToastHistory
+        ToastNotificationHistory History{ get; };
+
+        // Gets an Xml Payload based ona predefined Toast Template
+        Windows.Data.Xml.Dom.XmlDocument GetXmlTemplateContent(ToastTemplateType type);
+    };
+}
+```
+
+# Appendix
+
+-   To support cloud toasts, the developer will need to register the Toast Activator in
+    addition to the Push Activator registrations that are already being done. The Windows SDK will
+    inturn figure out the complexity of mapping cloud based toasts to the correct Toast Activator.
+-   Since building Toast XML payloads during runtime is non-trivial, we encourage developers to
+    utilize the windows developer toolkit, more specifically the ToastContentBuilder APIs to
+    construct the xml payload in the short term. Alternatively they can also use the ToastTemplates that WinAppSDK will provide to build their own toasts.
```

Comment:
```suggestion
-   Since building Toast XML payloads at runtime is non-trivial, we encourage developers to
    utilize the [Windows Community Toolkit](https://docs.microsoft.com/windows/communitytoolkit/), specifically the [ToastContentBuilder](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.uwp.notifications.toastcontentbuilder) APIs to
    construct the XML payload in the short term. Alternatively, they can also use the ToastTemplates that the Windows App SDK will provide to build their own toasts.
```